src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -184,4 +200,19 @@ void performArchival() throws InterruptedException, ExecutionException {
                         });
         return future;
     }
+
+    JsonArray getArchivedRecordings() throws InterruptedException, ExecutionException {
+        HttpRequest<Buffer> req = this.webClient.get("api/v1/recordings");
```

Comment:
We should avoid using HTTP requests to ourselves here - see #543 .

Reply:
Alright, so I should refactor the `RecordingsGetHandler` in such a way that the JSON array can be obtained without using an HTTP request? From a cursory glance at the handler, it uses `FileSystem` to store the data we need, so it looks like I could add and expose a method which extracts the `FileSystem` logic.

Reply:
Right. The logic should be extracted into something similar to the `RecordingCreationHelper`, for example called the `RecordingArchiveHelper`, which the `RecordingsGetHandler` and the `PeriodicArchiver` can both use for doing things related to adding/listing/deleting files from the archives. The `RecordingArchiveHelper` would take the existing logic from the `RecordingsGetHandler` that creates files in the `FileSystem` and generates their names, and that same logic but in reverse can be used to check the contents of the `FileSystem` and see if anything has a name that matches the format the `RecordingArchiveHelper` uses.

Reply:
Initially, when I saw the changes in @jan-law 's PR, I was thinking I would only have to implement listing archived files in my `RecordingArchiveHelper`, since adding and deleting files in the `PeriodicArchiver` has already been handled by the `RecordingHelper` from that PR.

However, I noticed that the `RecordingHelper` functions are also used by the recording handlers only concerning the target JVM such as `TargetRecordingDeleteHandler`, `TargetRecordingsPostHandler`, etc. 

Would it be worthwhile to keep the logic separate for archive vs. target recordings, i.e. have a `RecordingArchiveHelper` which the `PeriodicArchiver` and the archive related handlers (`TargetRecordingPatchSave`, `RecordingsGetHandler`, and `RecordingDeleteHandler`) all use for adding/listing/deleting files from the archives? 

Then the `RecordingHelper` (potentially renamed to `RecordingTargetHelper`) would no longer be used by the `PeriodicArchiver`, instead by recordings only involving the target?

Reply:
Sure, that does sound like a distinction probably worth making.

Reply:
@hareetd I can move my related helper functions `saveRecording` ,`deleteRecording` and `writeRecordingToDestination` into a new file `RecordingArchiveHelper`, and also rename `RecordingHelper` to `RecordingTargetHelper`. Then you can move your helper functions into `RecordingArchiveHelper`

Reply:
@jan-law Thank you, that sounds like it'll save me time re-writing the saving/deleting of the archive files. I'm just wondering how it'll work in terms of me getting the changes into my branch in a clean way (i.e. I still want my PR to be clear about its original intention) @andrewazores?

Reply:
@andrewazores since `public Optional<IRecordingDescriptor> getDescriptorByName` is used by both `RecordingTargetHelper` and `RecordingArchiveHelper`, should we make the function static? I think the function output only depends on its arguments.

Reply:
Rebase your PR on top of Janelle's branch instead of `main` :-) you can then edit your PR's text with "Depends on" and link it to hers that way.

Reply:
> @andrewazores since `public Optional<IRecordingDescriptor> getDescriptorByName` is used by both `RecordingTargetHelper` and `RecordingArchiveHelper`, should we make the function static? I think the function output only depends on its arguments.

Maybe each of the Helpers should have their own implementation of this function, returning a descriptor (if any) from their respective domains. If some consumer needs to get a descriptor from either domain by name then it can simply use both Helpers and take from them in whichever order of preference makes sense.